proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+syntax = "proto3";
+
+package redpanda.api.console.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Security {
+  enum PrincipalType {
+    PRINCIPAL_TYPE_UNSPECIFIED = 0;
+    PRINCIPAL_TYPE_USER = 1;
+    PRINCIPAL_TYPE_GROUP = 2;
+    PRINCIPAL_TYPE_REDPANDA_ROLE = 3;
+  }
+}
+
+// Role defienes a role in the system.
+message Role {
+  // The name of the role.
+  string name = 1;
+}
+
+// ListRolesRequest is the request for ListRoles.
+message ListRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+
+    // Return only roles assigned to this principal.
+    optional string principal = 2 [(buf.validate.field).string.max_len = 128];
+
+    // Return only roles of this principal type.
+    // Qualifies parameter principal has no effect on its own.
+    optional Security.PrincipalType principal_type = 3;
```

Comment:
I think generally it's considered best practice that if a field is optional semantically or conceptually to set it as such in the type as well. It communicated intent that it is not required. Furthermore it can be more performant as optional fields _when not set_ are not serialized and sent over the wire, and not parsed / deserialized on the other end.